compliation error

I am gettting:
```
error[E0716]: temporary value dropped while borrowed
  --> src/trainer/default/quant.rs:33:22
   |
15 |             let to_write: &[u8] = match self {
   |                 -------- borrow later stored here
...
33 |                     &x.to_le_bytes()
   |                      ^^^^^^^^^^^^^^^ creates a temporary value which is freed while still in use
34 |                 }
   |                 - temporary value is freed at the end of this statement
   |
   = note: consider using a `let` binding to create a longer lived value
```

--- a/src/trainer/default/quant.rs
+++ b/src/trainer/default/quant.rs
@@ -12,8 +12,8 @@ impl QuantTarget {
         let mut quantised = Vec::<u8>::new();
 
         for &float in buf {
-            let to_write: &[u8] = match self {
-                Self::Float => &float.to_le_bytes(),
+            let to_write = match self {
+                Self::Float => float.to_le_bytes().to_vec(),
                 Self::I16(q) => {
                     let x = (f32::from(q) * float) as i16;
 
@@ -21,7 +21,7 @@ impl QuantTarget {
                         return Err(io::Error::new(io::ErrorKind::InvalidData, "Failed quantisation from f32 to i16!"));
                     }
 
-                    &x.to_le_bytes()
+                    x.to_le_bytes().to_vec()
                 }
                 Self::I32(q) => {
                     let x = (q as f32 * float) as i32;
@@ -30,11 +30,11 @@ impl QuantTarget {
                         return Err(io::Error::new(io::ErrorKind::InvalidData, "Failed quantisation from f32 to i32!"));
                     }
 
-                    &x.to_le_bytes()
+                    x.to_le_bytes().to_vec()
                 }
             };
 
-            quantised.write_all(to_write)?;
+            quantised.write_all(&to_write)?;
         }
 
         Ok(quantised)

